src/runmode-dpdk.c

```diff
@@ -954,6 +954,25 @@ static void DeviceSetMTU(struct rte_eth_conf *port_conf, uint16_t mtu)
 #endif
 }
 
+/**
+ * @param port_id - queried port
+ * @param socket_id - socket ID of the queried port
+ * @return positive number on success, negative on failure (errno)
+ */
+static int32_t DeviceSetSocketID(uint16_t port_id, int32_t *socket_id)
+{
+    rte_errno = 0;
+    int retval;
+    retval = rte_eth_dev_socket_id(port_id);
+    *socket_id = retval;
+
+#if RTE_VERSION >= RTE_VERSION_NUM(22, 11, 0, 0)
+    retval = -rte_errno;
```

Comment:
seems weird that we first returned a value through `socket_id`

Reply:
and we still do if Suricata is compiled with <22.11 DPDK version. The change had to be made because DPDK API has changed. 

DPDK (latest)
https://doc.dpdk.org/api/rte__ethdev_8h.html#ad032e25f712e6ffeb0c19eab1ec1fd2e

DPDK (19.11)
https://doc.dpdk.org/api-19.11/rte__ethdev_8h.html#ad032e25f712e6ffeb0c19eab1ec1fd2e